pkg/utils/managedresources/builder/resourcemanager_test.go

```diff
@@ -229,7 +263,67 @@ var _ = Describe("Resource Manager", func() {
 					KeepObjects:                  pointer.Bool(keepObjects),
 					DeletePersistentVolumeClaims: pointer.Bool(deletePersistentVolumeClaims),
 				},
-			}))
+			}
+
+			Expect(references.InjectAnnotations(expectedMr)).To(Succeed())
+			Expect(mr).To(Equal(expectedMr))
+
+			for _, s := range secrets {
+				secret := &corev1.Secret{}
+				Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(s), secret)).To(Succeed())
+				expected := s.DeepCopy()
+				expected.ObjectMeta.ResourceVersion = "2"
+				expected.Labels["resources.gardener.cloud/garbage-collectable-reference"] = "true"
+				expected.TypeMeta = metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Secret",
+				}
+				Expect(secret).To(Equal(expected))
+			}
+		})
+
+		It("should label existing managed resource secrets", func() {
+			mr := &resourcesv1alpha1.ManagedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        name,
+					Namespace:   namespace,
+					Labels:      labels,
+					Annotations: annotations,
+				},
+				Spec: resourcesv1alpha1.ManagedResourceSpec{
+					SecretRefs: []corev1.LocalObjectReference{
+						{Name: secret1.Name},
+						{Name: secret2.Name},
+					},
+				},
+			}
+			Expect(fakeClient.Create(ctx, mr)).To(Succeed())
+			for _, s := range []*corev1.Secret{secret1, secret2, secret3} {
+				Expect(fakeClient.Create(ctx, s)).To(Succeed())
+			}
+
+			Expect(
+				NewManagedResource(fakeClient).
+					WithNamespacedName(namespace, name).
+					WithLabels(labels).
+					WithAnnotations(annotations).
+					WithSecretRef(secret1.Name).
+					WithSecretRef(secret2.Name).
```

Comment:
The test says "should label existing managed resource secrets" but in reality `WithSecretRef` would be called with the new, immutable secrets (i.e. name changes). Can you please adapt the test to really ensure, that the "old" secrets are marked as garbage collectable?

Reply:
Done